python/constraints.txt

```diff
@@ -371,10 +425,10 @@ pyarrow==14.0.2
     #   streamlit
 pycparser==2.21
     # via cffi
-pydantic==2.3.0
-    # via -r python/requirements.in
-pydantic-core==2.6.3
-    # via pydantic
+pydantic==1.10.13
```

Comment:
This is a big downgrade. Is there any way around this with the dbt stuff you're working on?

Also, there a reason to have this in this branch/PR rather than your dev branch?

Reply:
oh I didn't meant to modify this PR, I forgot that running the action manually uses the same branch name

I did these for 2 other PRs (PR list [here](https://github.com/NYCPlanning/data-engineering/pulls?q=is%3Apr+compile+python+requirements+)) that were merged into my dev branch

Reply:
Gotcha. Might make sense in general to not use this action for this, just compile requirements locally

Reply:
so far the only way around the pydantic downgrade is to downgrade dbt from 1.7 to 1.5

I looked through issues and PRs in the `dbt-semantic-interfaces` repo and pydantic 2.x support seems to be coming soon

Reply:
Hmm. Is semantic interfaces something you specifically want to use? Or just part of the later versions of dbt

Reply:
Not against downgrading pydantic if we need to, just curious

Reply:
unfortunately just part of later dbt versions, [seems like](https://www.getdbt.com/product/semantic-layer) it's all about BI metrics